### PR TITLE
feat: add renderer.highlight_hidden, renderer.icons.show.hidden and renderer.icons.hidden_placement for dotfile icons/highlights

### PR DESCRIPTION
**Pull request by evertonse**
#### This PR introduces the ability to highlight hidden files (dotfiles) by creating three new highlight groups:
```lua
  -- Hidden
  { group = "NvimTreeHiddenIcon", link = "NvimTreeSignColumn" },
  { group = "NvimTreeHiddenFileHL", link = "NvimTreeFileName" },
  { group = "NvimTreeHiddenFolderHL", link = "NvimTreeFolderName" },
```
It is ensured that if a file has a parent directory that is a dotfile, the file itself is also considered a dotfile. For example, all files within the ``.git`` or ``.github`` directories are regarded as dotfiles, as below:
image

#### To ensure backward compatibility, by default:

- Hidden files and folders highlight groups link to existing NvimTree highlights for files and folders.
- The hidden icon will not display by default.
- The name and icon of hidden files and folders will not be highlighted unless ``highlight_hidden`` is set to a value other than ``"none"``.
-  The ``show.hidden`` option is set to ``false`` by default.

Note: This change is not related to highlighting git-ignored files.

## Example Setup
```lua
require('nvim-tree').setup {
  renderer = {
    highlight_hidden = 'name',
    icons = {
      web_devicons = {
        file = {
          enable = true,
          color = true,
        },
        folder = {
          enable = false,
          color = true,
        },
      },
      hidden_placement = 'before',
      show = {
        hidden = true,
      },
      glyphs = {
        hidden = '󰜌',
      },
    },
  },
}
```

## Example Result (not generated by the above setup)
Below we show the highlights results produced by ``:Inspect`` command on a dotfile (hidden folder in this case).
image

### CODE REVIEW

**Comment by alex-courtis** [comment]
> ```lua
>   -- Hidden
>   { group = "NvimTreeHiddenIcon", link = "NvimTreeSignColumn" },
>   { group = "NvimTreeHiddenFileHL", link = "NvimTreeFileName" },
>   { group = "NvimTreeHiddenFolderHL", link = "NvimTreeFolderName" },
> ```

These defaults don't stand out very much; the user may think the feature is not working or misconfigured.

How about we use `Removed`? It's distinct and not already used.

```lua
  -- Hidden
  { group = "NvimTreeHiddenIcon", link = "Removed" },
  { group = "NvimTreeHiddenFileHL", link = "NvimTreeHiddenIcon" },
  { group = "NvimTreeHiddenFolderHL", link = "NvimTreeHiddenFileHL" },
```

File: lua/nvim-tree/renderer/builder.lua [hunk 1]
```diff
@@ -442,6 +443,7 @@ function Builder.setup(opts)
     DecoratorDiagnostics:new(opts),
     DecoratorBookmarks:new(opts),
     DecoratorModified:new(opts),
+    DecoratorHidden:new(opts),
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/builder.lua:446]: - [ ] Please update `:help nvim-tree.renderer.icons` with this order.

File: lua/nvim-tree/explorer/node.lua [hunk 2]
```diff
@@ -128,6 +128,20 @@ function M.is_git_ignored(node)
...
 
+---@param node Node
+---@return boolean
+function M.is_dotfile(node)
+  if node == nil then
+    return false
+  end
+  -- Inspect(node)
+  if node.is_dot or (node.name and (node.name:sub(1, 1) == ".")) or M.is_dotfile(node.parent) then
+    node.is_dot = true
```
Comment (alex-courtis) [inline lua/nvim-tree/explorer/node.lua:139]: Nice

**Comment by alex-courtis** [comment]
Test cases:

```lua
highlight_hidden = "none",
```

```lua
highlight_hidden = "icon",
```

```lua
highlight_hidden = "name",
```

```lua
highlight_hidden = "all",
```

```lua
      hidden_placement = "signcolumn",
      show = {
        hidden = true,
```

```lua
      hidden_placement = "before",
      show = {
        hidden = true,
```

```lua
      hidden_placement = "after",
      show = {
        hidden = true,
```

File: lua/nvim-tree/renderer/decorator/hidden.lua [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  return o
+end
+
+---Hidden icon: hidden.enable, renderer.icons.show.hidden and node starts with `.` (dotfile).
+---@param node Node
+---@return HighlightedString[]|nil icons
+function DecoratorHidden:calculate_icons(node)
+  if self.enabled and is_dotfile(node) then
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/decorator/hidden.lua:40]:
```suggestion
  if self.enabled and node:is_dotfile() then
```

You can then remove the require.

File: lua/nvim-tree/renderer/decorator/hidden.lua [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+---@return DecoratorHidden
+function DecoratorHidden:new(opts)
+  local o = Decorator.new(self, {
+    enabled = true,
+    hl_pos = HL_POSITION[opts.renderer.highlight_hidden] or HL_POSITION.none,
+    icon_placement = ICON_PLACEMENT[opts.renderer.icons.hidden_placement] or ICON_PLACEMENT.none,
+  })
+  ---@cast o DecoratorHidden
+
+  if not o.enabled then
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/decorator/hidden.lua:21]: - [ ] This will always be true, we can remove.

File: lua/nvim-tree/appearance/init.lua [hunk 5]
```diff
@@ -76,6 +76,11 @@ M.HIGHLIGHT_GROUPS = {
   { group = "NvimTreeModifiedFileHL", link = "NvimTreeModifiedIcon" },
   { group = "NvimTreeModifiedFolderHL", link = "NvimTreeModifiedFileHL" },
 
+  -- Hidden
+  { group = "NvimTreeHiddenIcon", link = "NvimTreeSignColumn" },
+  { group = "NvimTreeHiddenFileHL", link = "NvimTreeFileName" },
+  { group = "NvimTreeHiddenFolderHL", link = "NvimTreeFolderName" },
```
Comment (alex-courtis) [inline lua/nvim-tree/appearance/init.lua:82]: - [ ] Please add these to help `nvim-tree-highlight-default`

**Review by alex-courtis — CHANGES_REQUESTED** [review]
This is great - all working, just some polish needed.

File: lua/nvim-tree/renderer/decorator/hidden.lua (hunk 3 above)
Comment (evertonse) [reply lua/nvim-tree/renderer/decorator/hidden.lua]: I've updated in favor of all sugestions. However, I couldn't use ``is_dotfile`` as a method of ``node``. 
Instead I followed the same style of other decorators by 'requiring' into a variable called ``explorer_node`` and then using ``explorer_node.is_dotfile(node)``. 
Besides that, everything else should be aligned with the change request.

Comment (alex-courtis) [reply lua/nvim-tree/renderer/decorator/hidden.lua]: > However, I couldn't use `is_dotfile` as a method of `node`. Instead I followed the same style of other decorators by 'requiring' into a variable called `explorer_node` and then using `explorer_node.is_dotfile(node)`.

Right you are; node isn't yet a proper class. Apologies...

**Review by alex-courtis — APPROVED** [review]
Looks great, many thanks for your contribution!

**Pull request merged**